tensorflow/lite/micro/kernels/if.cc

```diff
@@ -34,13 +34,58 @@ namespace {
 struct OpData {
   int then_subgraph_index;
   int else_subgraph_index;
+  void* intermediate_copy_buffer;
+};
+
+enum class CopyInputDirection {
+  kFromIfToIntermediate,
+  kFromIntermediateToSubgraph,
+  kFromIntermediateToIf,
+};
+
+enum class CopyOutputDirection {
+  kFromSubgraphToIntermediate,
+  kFromIntermediateToIf,
 };
 
 void* Init(TfLiteContext* context, const char* buffer, size_t length) {
   TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
   return context->AllocatePersistentBuffer(context, sizeof(OpData));
 }
 
+size_t CalculateIntermediateBufferSize(TfLiteContext* context,
```

Comment:
If we do end up needing two separate buffers, this could be something like "AllocateIntermediateScratchBuffers" and allocate both input inermediate and output intermediate buffers.

This is quite a minor suggestion - feel free to ack and do whichever option you prefer.

Reply:
Thanks for the suggestion!  In the new PR, I rename this to AllocateIntermediateBuffers. I decide to avoid Scratch part because the name of Scratch buffer has been used extensively in other part of code and I feel quite confusing. I also want to call out that these intermediate buffers are in persistent arena since I could not find another way to guarantee that they don't overlap with in/output tensors (Pete and I discussed about this and we could not find a reasonable way to achieve this).